scripts/gb_model/compose_network.py

```diff
@@ -318,6 +318,91 @@ def add_pypsaeur_components(
     return n
 
 
+def process_demand_data(
+    demand_list: list[str],
+    clustered_demand_profile_list: list[str],
+    demand_type: list[str],
+    year: int,
+) -> pd.DataFrame:
+    """
+    Process the demand data for a particular demand type
+
+    Parameters
+    ----------
+    n : pypsa.Network
+        Network to finalize
+    demand_list: list[str]
+        CSV paths for demand data for each demand type
+    clustered_demand_profile_list: list[str]
+        CSV paths for demand shape for each demand type
+    demand_type:
+        demand type for which data is to be processed
+    year:
+        Year used in the modelling
+    """
+
+    # Filter file path for demand type
+    demand_path = [x for x in demand_list if demand_type in x][0]
+    demand_profile_path = [
+        x for x in clustered_demand_profile_list if demand_type in x
+    ][0]
+
+    # Read the files
+    demand = pd.read_csv(demand_path)
+    demand_profile = pd.read_csv(demand_profile_path, index_col=[0])
+
+    # Group demand data by year and bus and filter the data for required year
+    demand_grouped = demand.groupby(["year", "bus"]).sum().loc[year]
+
+    # Filtering those buses that are present in both the dataframes
+    list_of_buses = list(set(demand["bus"]) & set(demand_profile.columns))
+
+    # Scale the profile by the annual demand from FES
+    load = demand_profile[list_of_buses].mul(demand_grouped["p_set"])
+
+    # Convert load index to datetime dtype to avoid flagging an assertion error from pypsa
+    load.index = pd.to_datetime(load.index)
+
+    return load
+
+
+def add_load(
+    n: pypsa.Network,
+    demand_list: list,
+    clustered_demand_profile_list: list,
+    demand_types: list,
+    year: int,
+):
+    """
+    Add load as a timeseries to PyPSA network
+
+    Parameters
+    ----------
+    n : pypsa.Network
+        Network to finalize
+    demand_list: list[str]
+        CSV paths for demand data for each demand type
+    clustered_demand_profile_list: list[str]
+        CSV paths for demand shape for each demand type
+    demand_types:
+        Keywords to map the demand files to each demand type
+    year:
+        Year used in the modelling
+    """
+
+    # Iterate through each demand type
+    for demand_type in demand_types:
+        # Process data for the demand type
+        load = process_demand_data(
+            demand_list, clustered_demand_profile_list, demand_type, year
+        )
+
+        # Add the load to pypsa Network
+        suffix = f" {demand_type}"
+        n.add("Load", load.columns + suffix, bus=load.columns)
+        n.loads_t.p_set = pd.concat([n.loads_t.p_set, load.add_suffix(suffix)], axis=1)
```

Comment:
You should use our `_add_timeseries_data_to_network` to add this data.

EDIT: fixed that helper method to work in this case in #105